changefeedccl: add metamorphic testing for different schema changers

jayshrivastava opened the pull request:
This change updates some tests to use `maybeDisableDeclarativeSchemaChangesForTest()`
for metamorphic testing of behavior under both schema changers.

This change adds a test called `TestChangefeedSchemaChangeAllowBackfill_Legacy`
which is a carbon copy of `TestChangefeedSchemaChangeAllowBackfill` prior to
https://github.com/cockroachdb/cockroach/pull/107184. This legacy test only asserts behavior under the legacy schema changer.

This change also updates `maybeDisableDeclarativeSchemaChangesForTest()` to
not take a `rand.Rand` as a paramter. This is uncessessary as the global
`rand` pkg is sufficient for deterministic RNG during tests.

Closes: https://github.com/cockroachdb/cockroach/issues/107532
Closes: https://github.com/cockroachdb/cockroach/issues/106906
Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/108043)
<!-- Reviewable:end -->

jayshrivastava committed:
changefeedccl: add metamorphic testing for different schema changers

This change updates some tests to use `maybeDisableDeclarativeSchemaChangesForTest()`
for metamorphic testing of behavior under both schema changers.

This change adds a test called `TestChangefeedSchemaChangeAllowBackfill_Legacy`
which is a carbon copy of `TestChangefeedSchemaChangeAllowBackfill` prior to
https://github.com/cockroachdb/cockroach/pull/107184. This legacy test only asserts behavior under the legacy schema changer.

This change also updates `maybeDisableDeclarativeSchemaChangesForTest()` to
not take a `rand.Rand` as a paramter. This is uncessessary as the global
`rand` pkg is sufficient for deterministic RNG during tests.

Closes: https://github.com/cockroachdb/cockroach/issues/107532
Closes: https://github.com/cockroachdb/cockroach/issues/106906
Epic: None
Release note: None

jayshrivastava commented:
bors r=miretskiy
